components/metrics/builder.go

```diff
@@ -7,12 +7,28 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-func NewPrometheusMetricsBuilder(prometheusRegistry prometheus.Registerer, namespace string, subsystem string) PrometheusMetricsBuilder {
-	return PrometheusMetricsBuilder{
+type Option func(b *PrometheusMetricsBuilder)
+
+// CustomLabel allow to provide a custom metric label and a function to compute the value
+func CustomLabel(label string, fn LabelComputeFn) Option {
+	return func(b *PrometheusMetricsBuilder) {
+		b.customLabels = append(b.customLabels, metricLabel{
+			label:     label,
+			computeFn: fn,
+		})
+	}
+}
+
+func NewPrometheusMetricsBuilder(prometheusRegistry prometheus.Registerer, namespace string, subsystem string, opts ...Option) PrometheusMetricsBuilder {
```

Comment:
Hey @matdurand, sorry about the delay. I wanted to comment last week and just realized I didn't do it. 🤦‍♂️ 

I have just one comment regarding the API — most Watermill components use a "Config" struct instead of the Option pattern, and I feel it would be a good idea to be consistent, just for the improved developer experience. This means we would need another constructor, something like `NewPrometheusMetricsBuilderWithConfig`. What do you think?

Reply:
Hey @m110, yeah sure, consistency is paramount. I replaced Option with config. I also renamed "custom" for "additional", just because it sounded better imo. Let me know if these changes are enough. Thanks